ui.py

```diff
@@ -1464,27 +1464,31 @@ def draw_layer_source(context, layout, layer, layer_tree, source, image, vcol, i
                         boxcol.operator('node.y_set_decal_object_position_to_sursor', text='Set Position to Cursor', icon='CURSOR')
 
                     if layer.texcoord_type != 'Decal':
-                        mapping = get_layer_mapping(layer)
-
                         rrow = boxcol.row()
                         rrow.label(text='Transform:')
-                        rrow.prop(mapping, 'vector_type', text='')
+                        rrow.prop(get_layer_mapping(layer), 'vector_type', text='')
 
                         rrow = boxcol.row()
-                        if is_bl_newer_than(2, 81):
-                            mcol = rrow.column()
-                            mcol.prop(mapping.inputs[1], 'default_value', text='Offset')
-                            mcol = rrow.column()
-                            mcol.prop(mapping.inputs[2], 'default_value', text='Rotation')
-                            mcol = rrow.column()
-                            mcol.prop(mapping.inputs[3], 'default_value', text='Scale')
+                        mcol = rrow.column()
+                        mcol.prop(layer, 'translation', text='Offset')
+                        mcol = rrow.column()
+                        mcol.prop(layer, 'rotation', text='Rotation')
```

Comment:
Both Layer and Mask already have translation/rotation/offset properties that go through `update_mapping` so it makes sense to use them in UI props. It's less LOC and makes it easier to set the uniform scale float to min scale component when the uniform option is enabled

Reply:
Those properties are actually from the old implementations, now all ucupaint UI should come from actual nodes, not from python properties. This is because python properties do not work with animation and driver.